src/main/java/ru/berdinskiybear/armorhud/mixin/InGameHudMixin.java

```diff
@@ -78,37 +83,44 @@ private void drawArmorHud(DrawContext context, RenderTickCounter tickCounter) {
 
         if (config.getOrientation() == ArmorHudConfig.Orientation.VERTICAL) {
-            context.getMatrices().rotate(org.joml.Math.toRadians(90f)).translate(0, -22);
+            context.getMatrices().rotate(MathHelper.HALF_PI).translate(0, -SIZE);
         }
 
         switch (config.getStyle()) {
             case HOTBAR -> {
-                context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, InGameHud.HOTBAR_TEXTURE, 182, 22, 0, 0, 0, 0, textureWidth - 3, HEIGHT);
-                context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, InGameHud.HOTBAR_TEXTURE, 182, 22, 182 - 3, 0, textureWidth - 3, 0, 3, HEIGHT);
+                context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, HOTBAR_TEXTURE, 182, 22, 0, 0, 0, 0, textureWidth - 3, SIZE);
+                context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, HOTBAR_TEXTURE, 182, 22, 182 - 3, 0, textureWidth - 3, 0, 3, SIZE);
             }
             case ROUNDED_CORNERS -> {
-                context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, InGameHud.HOTBAR_OFFHAND_LEFT_TEXTURE, 29, 24, 0, 1, 0, 0, 3, HEIGHT);
-                context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, InGameHud.HOTBAR_TEXTURE, 182, 22, 3, 0, 3, 0, textureWidth - 6, HEIGHT);
-                context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, InGameHud.HOTBAR_OFFHAND_LEFT_TEXTURE, 29, 24, WIDTH - 3, 1, textureWidth - 3, 0, 3, HEIGHT);
+                if (armorItems.size() > 1) {
+                    context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, HOTBAR_OFFHAND_LEFT_TEXTURE, 29, 24, 0, 1, 0, 0, 3, SIZE);
+                    context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, HOTBAR_TEXTURE, 182, 22, 3, 0, 3, 0, textureWidth - 6, SIZE);
+                    context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, HOTBAR_OFFHAND_LEFT_TEXTURE, 29, 24, SIZE - 3, 1, textureWidth - 3, 0, 3, SIZE);
+                } else {
+                    context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, HOTBAR_OFFHAND_LEFT_TEXTURE, 29, 24, 0, 1, 0, 0, SIZE, SIZE);
+                }
             }
             case ROUNDED -> {
-                int borderWidth = (WIDTH - STEP) / 2;
-                context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, InGameHud.HOTBAR_OFFHAND_LEFT_TEXTURE, 29, 24, 0, 1, 0, 0, borderWidth, HEIGHT);
-                for (int i = 0; i < armorItems.size(); i++) {
-                    context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, InGameHud.HOTBAR_OFFHAND_LEFT_TEXTURE, 29, 24, borderWidth, 1, borderWidth + i * STEP, 0, STEP, HEIGHT);
+                if (armorItems.size() > 1) {
+                    int borderWidth = (SIZE - STEP) / 2;
+                    context.drawGuiTexture(RenderPipelines.GUI_TEXTURED, HOTBAR_OFFHAND_LEFT_TEXTURE, 29, 24, 0, 1, 0, 0, SIZE - borderWidth, SIZE);
+                    // nothing happens if slots <= 2
+                    for (int i = armorItems.size() - 2; i >= 1; i--) {
```

Comment:
Why the decreasing for loop? Wouldn't something like this work?
```suggestion
                    for (int i = 0; i < armorItems.size() - 2; i++) {
```

Reply:
it would, but `i` would need to be initialized to 1